src/main/java/com/webapp/TextBook/controller/web_controller/HomeController.java

```diff
@@ -1,42 +1,118 @@
+    public String login(){ return "login"; }
+
+    //login confirmation
+    @RequestMapping(value = "/loginConfirmed", method = RequestMethod.GET)
+    public String loginConfirmation (
+            @Valid @ModelAttribute("LoginUserInfo") LoginUserInfo person,
+            BindingResult result,
+            ModelMap map)  {
+
+        // handling all exceptions with try catch block
+        try {
+
+            // format validation/dealing with binding errors
+            final String failedValidationStatusMessage = "User input missing or invalid";
+            if (result.hasErrors()) {
+                //binding errors present, load in Status/Error messages and return to login
+
+                assert (result.getErrorCount() != 0);
+                ObjectError loginUserInfoError = result.getAllErrors().get(0);
+                map.addAttribute("Error", new JSONArray(loginUserInfoError.getDefaultMessage()));
+                map.addAttribute("StatusMessage", failedValidationStatusMessage);
+                map.addAttribute("User", null);
+
+                return "login";
+            }
+
+            // checking user data validity with database
+            Pair<Optional<User>, StatusCode> user = adapter.userLogin(person.get_username(), person.get_password());
+
+            //interpreting data
+            if (user.getValue1() == StatusCode.OK) {
+                // User login was valid and will be further processed
+
+                map.addAttribute("Error", null);
+                map.addAttribute("StatusMessage", user.getValue1().getContentMessage());
+                map.addAttribute("User", person);
+
+                //setting user session data
+                SharedSessionData.setSessionValueWithKey(SharedSessionData.USER_KEY, user.getValue0().orElseThrow());
+
+                //checking user Role
+                if (user.getValue0().get().userRole == UserRole.Supervisor) {
+                    // User Role is Supervisor and will continue as such
+                    return "supervisorDropdown";
+                } else {
+                    // User Role is Student Employee and will continue as such
+                    return "studentDropdown";
+                }
+            } else {
+                // User login was invalid for reason stated in status message
+                // return to login
+                map.addAttribute("Error", null);
+                map.addAttribute("StatusMessage", user.getValue1().getContentMessage());
+                map.addAttribute("User", null);
+
+                return "login";
+            }
+        }
+        catch (Exception e){
+            //Catching Exceptions and printing StackTrace of the error
+            //Loading in custom status message and returning to login page
+            System.out.println(e.getStackTrace());
+            map.addAttribute("Error", null);
+            map.addAttribute("StatusMessage", "Internal error has occurred. " +
```

Comment:
The FitnessGram Pacer Test is a multistage aerobic capacity test that progressively gets more difficult as it continues. The 20 meter pacer test will begin in 30 seconds. Line up at the start. The running speed starts slowly but gets faster each minute after you hear this signal bodeboop. A sing lap should be completed every time you hear this sound. ding Remember to run in a straight line and run as long as possible. The second time you fail to complete a lap before the sound, your test is over. The test will begin on the word start. On your mark. Get ready!… Start.

Reply:
https://github.com/NicholasPierce1/TextBookServices/commit/fefa1bcf3c2b9de1242bee44ff1911d469e5c366